android-sdk/src/main/java/org/sagebionetworks/bridge/android/viewmodel/PhoneAuthViewModel.java

```diff
@@ -0,0 +1,191 @@
+                            LOGGER.debug("Session renewal succeeded.");
+                            // current session is still useable
+                            signInStateLiveData.postValue(Resource.success(new Object()));
+                        }, t -> {
+                            LOGGER.debug("Session renewal failed, signing in.");
+                            doPhoneSignIn(token, phoneRegion, phoneNumber);
+                        }));
+    }
+
+    @VisibleForTesting
+    boolean isNewSignIn(final String phoneRegion, final String phoneNumber) {
+        final boolean newSignIn;UserSessionInfo currentSession = authenticationManager.getUserSessionInfo();
+        if (currentSession == null) {
+            newSignIn = true;
+        } else {
+            // TODO: may not be reliable since session does not always contain phone
+            // but that just means we will sign in again
+            Phone phone = currentSession.getPhone();
+            newSignIn = !isSamePhone(phone, phoneRegion, phoneNumber);
+        }
+        return newSignIn;
+    }
+
+    @VisibleForTesting
+    boolean isSamePhone(Phone expected, String phoneRegion, String phoneNumber) {
+        return expected != null && expected.getRegionCode().equals(phoneRegion)
+                && phoneNumber != null && expected.getNumber().endsWith(phoneNumber);
+    }
+
+    @VisibleForTesting
+    void doPhoneSignIn(String token, String phoneRegion, String phoneNumber) {
+        if (Strings.isNullOrEmpty(token)) {
+            LOGGER.error("Received null or empty token");
+
+            signInStateLiveData.postValue(
+                    Resource.error("Unable to process sign in link. Please sign in again to get a new link by SMS",
+                            new Object()));
+            return;
+        }
+
+        if (Strings.isNullOrEmpty(phoneRegion) || Strings.isNullOrEmpty(phoneNumber)) {
+            LOGGER.warn("Phone number and region are required and were not found in accountDAO");
+
+            signInStateLiveData.postValue(
+                    Resource.error("Phone number unknown. Please sign in again to get a new link by SMS",
+                            new Object()));
+            return;
+        }
+
+        compositeSubscription.add(authenticationManager
+                .signInViaPhoneLink(phoneRegion, phoneNumber, token)
+                .subscribe(session -> {
+                    LOGGER.debug("TokenLoginSubscribe: Authenticated Login Complete");
+                    signInStateLiveData.postValue(Resource.success(new Object()));
+                }, throwable -> {
+                    LOGGER.warn("Sign up failed", throwable);
+
+                    signInStateLiveData.postValue(
+                            Resource.error("Error signing in: \n" + throwable.getMessage(),
+                                    new Object()));
+                }));
+    }
+
+    @Override
+    protected void onCleared() {

```

Comment:
do you need super.onCleared()?

Reply:
probably safer, even tho super iks a no-op